csharp/ql/test/library-tests/expressions/expressions.cs

```diff
@@ -432,6 +432,18 @@ void MainCreations()
         delegate int S(int x, int y);
         delegate void Unit();
 
+        void MultiDimensionalArrayCreations()
+        {
+            object o = new int[,] { { 1, 2 }, { 3, 4 }, { 5, 6 } };
+            o = new int[,,] { { { 1, 2, 3 }, { 4, 5, 6 } }, { { 7, 8, 9 }, { 10, 11, 12 } } };
+            o = new int[,][,]
```

Comment:
For this array creation, `getLengthArgument(0) = 2` and `getLengthArgument(1) = 3`, which means we only get length arguments for the outer array type. I wonder if we should instead have two predicates on the `ArrayCreation` class:
```
int getDimension()
Expr getLengthArgument(int dim, int i)
```
where if we modify the example slightly to
```csharp
            o = new int[,][,]
            {
                { new int[,] { {1}, {5} }, new int[,] { {0}, {4} }, new int[,] { {11}, {88} }, new int[,] { {11}, {0} } },
                { new int[,] { {1}, {5} }, new int[,] { {0}, {4} }, new int[,] { {11}, {88} }, new int[,] { {11}, {0} } },
                { new int[,] { {1}, {5} }, new int[,] { {0}, {4} }, new int[,] { {11}, {88} }, new int[,] { {11}, {0} } }
            };
```
we would have
```
getDimension() = 2
getLengthArgument(0, 0) = 1
getLengthArgument(0, 1) = 2
getLengthArgument(1, 0) = 4
getLengthArgument(1, 1) = 3
```
same if we had used explicit sizes:
```
o = new int[1,2][4,3]
```

Reply:
I'm not sure I understand your suggestion. But I don't think this would work. `new int[1,2][4,3]` is not valid in C#. Lengths can only be specified in the first `RankSpecifier`.
There's no way to guarantee that the non-first array rank specifiers have the same dimensions. The following two are the same:
```
new int[][,] { new int[1, 2], new int[3, 4] }
new int[2][,] { new int[1, 2], new int[3, 4] }
```
and you can't specify it as `new int[2][X,Y] {...}`.

Reply:
> `new int[1,2][4,3]` is not valid in C#. Lengths can only be specified in the first `RankSpecifier`.

Indeed, you are right. Sorry about the confusion.